File: backend/src/main/java/com/concessionaria/auth/AuthService.java
package com.concessionaria.auth;

import com.concessionaria.auth.dto.LoginRequest;
import com.concessionaria.auth.dto.LoginResponse;
import com.concessionaria.exception.ResourceNotFoundException;
import com.concessionaria.security.JwtTokenProvider;
import com.concessionaria.user.User;
import com.concessionaria.user.UserRepository;
import com.concessionaria.user.dto.UserResponse;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional(readOnly = true)
public class AuthService {

    private final AuthenticationManager authenticationManager;
    private final UserRepository userRepository;
    private final JwtTokenProvider jwtTokenProvider;

    public AuthService(AuthenticationManager authenticationManager, UserRepository userRepository,
                        JwtTokenProvider jwtTokenProvider) {
        this.authenticationManager = authenticationManager;
        this.userRepository = userRepository;
        this.jwtTokenProvider = jwtTokenProvider;
    }

    public LoginResponse login(LoginRequest request) {
        authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(request.email(), request.password()));

        User user = userRepository.findByEmailIgnoreCase(request.email())
                .orElseThrow(() -> ResourceNotFoundException.of("Usuário", request.email()));

        String token = jwtTokenProvider.generateToken(user.getEmail(), user.getRole().name(), user.getName());
        return new LoginResponse(token, "Bearer", jwtTokenProvider.getExpirationMs(), UserResponse.from(user));
    }

    public UserResponse me(String email) {
        User user = userRepository.findByEmailIgnoreCase(email)
                .orElseThrow(() -> ResourceNotFoundException.of("Usuário", email));
        return UserResponse.from(user);
    }
}

File: frontend/src/services/authService.ts
import { apiClient } from "./apiClient";
import type { User } from "@/types/user";

export interface LoginValues {
  email: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  tokenType: string;
  expiresInMs: number;
  user: User;
}

export const authService = {
  async login(values: LoginValues): Promise<LoginResponse> {
    const { data } = await apiClient.post<LoginResponse>("/auth/login", values);
    return data;
  },

  async me(): Promise<User> {
    const { data } = await apiClient.get<User>("/auth/me");
    return data;
  },
};
